Refactor/props and rename api functions

nnyouung opened the pull request:
### 개요
api 변수명 직관적으로 변경 및 props 관련 수정 (PR #74 피드백)
<br>

### 수정사항
- {props} 대신 명시적인 파라미터명으로 변경
- api 함수명 변경 및 좋아요 관련 api를 게시글 및 댓글용으로 구분

보다 자세한 내용은 커밋 확인 부탁드립니다! 시각적인 변화는 없어 테스트 화면은 생략하도록 하겠습니다.
<br>

nnyouung committed:
feat: connectPage 검색 기능 백엔드 연결

수정사항
- api.js, ConnectPage.js
  - connectPage 검색 기능 백엔드 연결
- ConnectPage.js
  - 키보드의 검색 버튼을 눌러도 handleSearch가 호출되도록 onSubmitEditing 추가

nnyouung committed:
feat: 필터 기능 백엔드 연결 및 리셋 버튼 클릭 시 검색어 삭제 및 검색 결과 초기화

수정사항
- api.js, ConnectPage.js, FilterBottomSlide.js
  - 필터 기능 백엔드 연결
- ConnectPage.js
  - 리셋 버튼 클릭 시 검색창의 내용을 비우고, 검색 결과를 초기화하여 데이터를 다시 불러옴

nnyouung committed:
feat: 검색 및 필터 검색 오류 발생 시의 컴포넌트 추가

수정사항
- searchFail.svg, IconSearchFail.jsx
  - 컴포넌트 제작을 위한 아이콘 생성
- ConnectCard.js, FilterBottomSlide.js, ConnectPage.js
  - 검색 및 필터 오류 반환 시, ‘일치하는 검색 결과가 없습니다’ 컴포넌트 띄우도록 수정
- ConnectStyles.js
  - 세부 스타일 변경

nnyouung committed:
style: Prettier & ESLint 적용

nnyouung committed:
refector: {props} 대신 명시적인 파라미터명으로 변경

수정사항
- 가독성을 높이기 위해, {props} 대신 개별적인 속성을 이용하여 파리미터명 변경

nnyouung committed:
refector: api 함수명 변경 및 좋아요 관련 api를 게시글 및 댓글용으로 구분

수정사항
- api.js, LoginPage.js, MemberPage.jsx, ModifyProfilePage.jsx, NicknamePage.jsx, BookmarkPostPage.jsx, StudentVerificationPage.jsx
  - api 함수명을 직관적으로 변경
- api.js, PostPage.jsx, ItemComment.jsx
  - 좋아요 관련 api를 게시글 및 댓글용으로 구분

nnyouung committed:
style: Prettier & ESLint 적용2

nnyouung committed:
style: Prettier & ESLint 적용

seungholee-dev commented on src/components/community/IconHeart.jsx:
```diff
@@ -2,7 +2,7 @@ import * as React from "react";
 import { CustomTheme } from "@styles/CustomTheme";
 
-const IconHeart = ({ props, active }) => {
+const IconHeart = ({ active, ...props }) => {
```
전체를 props로 사용하시거나, props의 항목을 destructuring을 통해 빼내야할 것 같네요!

nnyouung replied:
피드백 감사합니다!

아이콘의 경우에는 active처럼 코드 내에서 로직에 이용되지 않는 한, style이나 onpress 같은 일반적인 props는 자유롭게 사용하려고 '...props'로 전달했는데요. 디자인이 완성되지 않은 상태에서 만들어가다보니, 같은 아이콘이 예측할 수 없는 여러 방식으로 쓰이게 돼서 그 편이 재사용성이 더 좋다고 생각하였습니다!

아이콘을 제외한 다른 컴포넌트에서의 props는 destructuring을 통해 빼내는 방식을 선택했는데, 재사용성을 고려하더라도 다른 컴포넌트들과 똑같이 구현하는 게 나을까요?

seungholee-dev replied:
이따가 회의할 때 진행하시죠! :)

seungholee-dev commented on src/pages/onboarding/NicknamePage.js:
```diff
@@ -18,7 +18,7 @@ import Progress1 from "@components/onboarding/Progress1";
 import IconDelete from "@components/onboarding/IconDelete";
 import ApplyButton from "@components/common/ApplyButton";
-import { headCheckUserName } from "config/api";
+import { checkUserName } from "config/api";
```
`checkUsername`이 좋을 것 같아요!

seungholee-dev commented on src/pages/member/BookmarkPostPage.jsx:
```diff
@@ -3,7 +3,7 @@ import { View, ScrollView } from "react-native";
 import BookmarkPostStyles from "@pages/member/BookmarkPostStyles";
 
-import { getBookmarkPost } from "config/api";
+import { getBookmarkedPost } from "config/api";
```
`getBookmarkedPosts`로 복수형으로 써야겠네요!

nnyouung replied:
동시에 getLikedPost도 복수형으로 바꿔놓았습니다!

seungholee-dev reviewed (approved):
고생하셨습니다!

nnyouung committed:
style: api 변수명 수정

수정사항
- api.js, PostPage.jsx, LikedPostPage.jsx, BookmarkPostPage, NicknamePage.js
  - api 변수명을 복수형으로 수정 (getBookmarkedPost -> getBookmarkedPosts, getLikedPost -> getLikedPosts)
